packages/lodestar/src/util/queue.ts

```diff
@@ -37,6 +38,9 @@ export class JobQueue {
     this.processor = pipe(this.queue, async (source) => {
       for await (const job of source) {
         await this.processJob(job);
+        // in case we have adjacent events in the event loop of the same job like block processing
+        // we want other event to have a chance to execute too
+        await sleep(0);
```

Comment:
right now we use JobQueue for block processing and for regen module. I think both of them need this trick.

Reply:
We should benchmark the performance implications of this. Maybe it's not necessary to do this on every block but every N blocks

Reply:
I don't think it has a big performance impact as this basically means we put the execution of this event to the end of the event loop. Jobs putting to this JobQueue is meant to be heavy already, others should be lightweight or io operations and it'll be executed soon.

But I also have some statistics in my local environment for the current performance of block processing: processing a normal block in regular sync takes 1.5s-2s in average, processing an epoch takes 4s-5s.